portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,20 @@ function getRandomQuote() {
   const quoteContainer = document.getElementById('quote-container');
   quoteContainer.innerText = random_quote;
 }
+
+/** Fetches hardcoded string from the server and adds it to the page. */
+async function showRandomFact() {
+    // Fetch JSON from server
```

Comment:
You want comments which communicate to the reader things that are not immediately obvious when reading through the code. 
As a toy example, take: 
```javascript
// Print hello
console.log("hello");
```
The comment didn't communicate anything new to the reader and the reader had to parse through both lines. 

On the other hand, this fun little piece of coding history trivia has comments which provide great value: 
```cpp
 // A fast square root finding approximate algorithm. 
 float InvSqrt(float x){
        // See https://en.wikipedia.org/wiki/Fast_inverse_square_root for more details.
        float xhalf = 0.5f * x;
        int i = *(int*)&x;            // store floating-point bits in integer
        i = 0x5f3759df - (i >> 1);    // initial guess for Newton's method
        x = *(float*)&i;              // convert new bits into float
        x = x*(1.5f - xhalf*x*x);     // One round of Newton's method
        return x;
    }
```